File: debfile.py
#!/usr/bin/python3

import sys
import dbm.gnu

database = "/var/cache/debfile.db"
query = sys.argv[1]

dbm = dbm.gnu.open(database, 'r', 0o444)
try:
    pkg = dbm[query].decode('ascii')
    print("%s_%s" % (pkg, dbm[pkg].decode('ascii')))
except KeyError:
    sys.exit(1)

File: debfile.pl
#!/usr/bin/perl

use strict;
use GDBM_File;

my $database='/var/cache/debfile.db';
my $q = $ARGV[0];

my %dbm;
tie %dbm, 'GDBM_File', $database, GDBM_READER, 0444
    or die "$GDBM_File::gdbm_errno";
my $pkg = $dbm{$q};
print "$dbm{$pkg}\n" if $pkg;
untie %dbm;
exit 1 unless $pkg;
